lib/src/utils.dart

```diff
@@ -399,3 +412,33 @@ void ensureJsonEncodable(Object message) {
     throw new ArgumentError.value("$message can't be JSON-encoded.");
   }
 }
+
+/// Prepends each line in [text] with [prefix].
+///
+/// If [first] or [last] is passed, the first and last lines, respectively, are
+/// prefixed with those instead. If [single] is passed, it's used if there's
+/// only a single line; otherwise, [first], [last], or [prefix] is used, in that
+/// order of precedence.
+String prefixLines(String text, String prefix, {String first, String last,
+    String single}) {
+  first ??= prefix;
+  last ??= prefix;
+  single ??= first ?? last ?? prefix;
+
+  var lines = text.split('\n');
+  if (lines.length == 1) return "$single$text";
+
+  var buffer = new StringBuffer("$first${lines.first}\n");
+  for (var line in lines.skip(1).take(lines.length - 2)) {
```

Comment:
this line with the skip and take  len - 2 is a little complex. Consider adding a comment describing why it's like this